interfaces/builtin/storage_framework_test.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	//"github.com/snapcore/snapd/testutil"
+)
+
+type StorageFrameworkInterfaceSuite struct {
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&StorageFrameworkInterfaceSuite{
+	iface: &builtin.StorageFrameworkInterface{},
+	slot: &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "server", Type: snap.TypeOS},
+			Name:      "storage-framework",
+			Interface: "storage-framework",
+		},
+	},
+	plug: &interfaces.Plug{
+		PlugInfo: &snap.PlugInfo{
+			Snap:      &snap.Info{SuggestedName: "client"},
+			Name:      "storage-framework",
+			Interface: "storage-framework",
+		},
+	},
+})
+
+func (s *StorageFrameworkInterfaceSuite) TestName(c *C) {
+	c.Check(s.iface.Name(), Equals, "storage-framework")
+}
+
+func (s *StorageFrameworkInterfaceSuite) TestSanitizeIncorrectInterface(c *C) {
+	c.Check(func() { s.iface.SanitizeSlot(&interfaces.Slot{SlotInfo: &snap.SlotInfo{Interface: "other"}}) },
+		PanicMatches, `slot is not of interface "storage-framework"`)
+	c.Check(func() { s.iface.SanitizePlug(&interfaces.Plug{PlugInfo: &snap.PlugInfo{Interface: "other"}}) },
+		PanicMatches, `plug is not of interface "storage-framework"`)
+}
+
+func (s *StorageFrameworkInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	// connected slots have a non-nil security snippet for apparmor
+	snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+	// slots have a permanent non-nil security snippet for apparmor
+	snippet, err = s.iface.PermanentSlotSnippet(s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+	// plugs have a non-nil security snippet for apparmor
+	snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+}
```

Comment:
It would be nice to add a test that the rules you are adding for ConnectedSlot and ConnectedPlug policy are preset similar to thumbnailer_test.go in TestSlotGrantedAccessToPlugFiles().